core/domain/feedback_services.py

```diff
@@ -368,7 +421,123 @@ def _get_thread_from_model(thread_model):
+
+
+def set_message_count_for_thread(full_thread_id, message_count):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count = message_count
+    thread_model.put()
+
+
+def increase_message_count_by_one(full_thread_id):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count += 1
+    thread_model.put()
+
+
+def get_thread_summaries(user_id, full_thread_ids):
+    """Returns a list of summaries corresponding to each of the threads given.
+
+    Args:
+        thread_ids: list(str). The ids of the threads excluding the exploration
+            id part.
+        exploration_ids: list(str). The ids of the explorations to which the
+            threads belong.
+
+    Returns:
+        list(dict). A list of dictionary containing the summaries of the threads
+            given to it.
+    """
+    exploration_and_thread_ids = (
+        [thread_id.split('.') for thread_id in full_thread_ids])
+    exploration_ids, thread_ids = zip(*exploration_and_thread_ids)
+
+    thread_model_ids = (
+        [feedback_models.FeedbackThreadModel.generate_full_thread_id(
+            exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    feedback_thread_user_model_ids = (
+        [feedback_models.FeedbackThreadUserModel.generate_full_id(
+            user_id, exploration_id, thread_id)
+         for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+
+    last_two_messages_ids = (
+        feedback_models.FeedbackMessageModel.get_last_two_message_ids_of_threads( # pylint: disable=line-too-long
+            exploration_ids, thread_ids))
+
+    multiple_models = (
+        datastore_services.fetch_multiple_entities_by_ids_and_models(
+            [
+                ('FeedbackThreadModel', thread_model_ids),
+                ('FeedbackThreadUserModel', feedback_thread_user_model_ids),
+                ('ExplorationModel', exploration_ids),
+                ('FeedbackMessageModel', last_two_messages_ids)
+            ]))
+
+    thread_models = multiple_models[0]
+    feedback_thread_user_models = multiple_models[1]
+    explorations = multiple_models[2]
+    messages = multiple_models[3]
+
+    last_two_messages = [messages[i:i + 2] for i in range(0, len(messages), 2)]
+
+    thread_summaries = []
+    for index, model in enumerate(thread_models):
+
+        does_second_message_exist = (last_two_messages[index][1] is not None)
```

Comment:
Nit: might as well put this directly above line 512 so that there's less context that the reader needs to remember.